### PR TITLE
[MSHARED-1286] Display if dependency is optional in tree

### PR DESCRIPTION
**Pull request by vbreivik**
This should reintroduce that dependency:tree shows if a dependency is optional which was removed in 3.3.0 of dependency plugin.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/MSHARED) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MSHARED-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MSHARED-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
I have created an issue for it https://issues.apache.org/jira/browse/MSHARED-1286
- please change commit message
- please select at least `I hereby declare this contribution to be licenced...` in PR description

**Comment by vbreivik** [comment]
> I have created an issue for it https://issues.apache.org/jira/browse/MSHARED-1286
> 
> * please change commit message
> * please select at least `I hereby declare this contribution to be licenced...` in PR description

Commit message is updated and box ticked. :)

File: pom.xml [hunk 1]
```diff
@@ -96,6 +96,13 @@
       <version>1</version>
       <optional>true</optional>
     </dependency>
+
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.13.2</version>
+      <scope>test</scope>
+    </dependency>
```
Comment (slawekjaranowski) [inline pom.xml:105]: Please use JUnit 5

File: src/test/java/org/apache/maven/shared/dependency/graph/internal/DefaultDependencyNodeTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * under the License.
+ */
+package org.apache.maven.shared.dependency.graph.internal;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.DefaultArtifact;
+import org.junit.Test;
+
+import static java.util.Collections.emptyList;
+import static org.junit.Assert.*;
```
Comment (slawekjaranowski) [inline src/test/java/org/apache/maven/shared/dependency/graph/internal/DefaultDependencyNodeTest.java:26]: Please don't use asterisk for imports

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: src/test/java/org/apache/maven/shared/dependency/graph/internal/DefaultDependencyNodeTest.java (hunk 2 above)
Comment (vbreivik) [reply src/test/java/org/apache/maven/shared/dependency/graph/internal/DefaultDependencyNodeTest.java]: Whops, I am used to build breaking when I forget. 
Fixed.

File: pom.xml [hunk 3]
```diff
@@ -96,6 +96,13 @@
       <version>1</version>
       <optional>true</optional>
     </dependency>
+
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
```
Comment (slawekjaranowski) [inline pom.xml:102]: `junit-jupiter-api` will be good enough here - we don't need engine in test classpath

**Review by slawekjaranowski — APPROVED** [review]

**Review by cstamas — APPROVED** [review]

**Pull request merged**